internal/service/alarms/api/server.go

```diff
@@ -363,13 +370,37 @@ func (a *AlarmsServer) GetProbableCause(ctx context.Context, request api.GetProb
 	return nil, fmt.Errorf("not implemented")
 }
 
+// AmNotification handles an API request coming from AlertManager with CaaS alerts. This api is used internally.
+// Note: the errors returned can also be view under alertmanager pod logs but also logging here for convenience
 func (a *AlarmsServer) AmNotification(ctx context.Context, request api.AmNotificationRequestObject) (api.AmNotificationResponseObject, error) {
-	// TODO: Implement the logic to handle the AM notification
-	slog.Debug("Received AM notification", "groupLabels", request.Body.GroupLabels)
-	for _, alert := range request.Body.Alerts {
-		slog.Debug("Alert", "fingerprint", alert.Fingerprint, "startsAt", alert.StartsAt, "status", alert.Status)
+	// Audit the table with full list of alerts in the current payload. If missing set them to resolve
+	if err := a.AlarmsRepository.ResolveNotificationIfNotInCurrent(ctx, request.Body); err != nil {
+		msg := "failed to resolve notification that are not present"
+		slog.Error(msg, "error", err)
+		return nil, fmt.Errorf("%s: %w", msg, err)
+	}
+
+	// Get the definition data based on current set of Alert names and managed cluster ID
+	alarmDefinitions, err := a.AlarmsRepository.GetAlarmDefinitions(ctx, request.Body, a.ClusterServer.ClusterIDToResourceTypeID)
+	if err != nil {
+		msg := "failed to get AlarmDefinitions"
+		slog.Error(msg, "error", err)
+		return nil, fmt.Errorf("%s: %w", msg, err)
 	}
 
+	// Combine possible definitions with events
```

Comment:
Please update cleared time, changed time and perceived severity of resolved alarms in code. Discussed this with @alegacy  on previous prs where I tried to add some triggers. The conclusion was that triggers are difficult to maintain and we should avoid them. It makes the logic more difficult to follow.

Reply:
I'm not being pushing. If you don't want to add it, it's fine. But please just for my understanding because it seems I'm missing something.

The `change_time` is (copying from the spec) a Date/Time stamp value when any value of the AlarmEventRecord has been modified. The AlarmEventRecord after being added the first time, can only be modified when it is acknowledged or cleared. The acknowledge operation comes from the API. There, we update the change time. For Manual clearing type alarms, the operation also comes from the API. There, we also update the change time. Now, for this part, Automatic clearing type alarms, we are not setting it when we clear the alarm. Why? What is preventing us to do it? Again, it is just for my understanding not trying to impose anything.

And I don't think this value should come from AlertManager, the spec clearly says that it is only for values of the alarm record. In fact, as far as I know, the alertManager only updates an alert when it changes the firing status to Resolved. The EndsAt is also updated.

c.c. @browsell in case he has more info about this constraint.